pkg/daemon/ceph/client/crush.go

```diff
@@ -202,6 +202,10 @@ func GetOSDOnHost(context *clusterd.Context, clusterInfo *ClusterInfo, node stri
 	args := []string{"osd", "crush", "ls", node}
 	buf, err := NewCephCommand(context, clusterInfo, args).Run()
 	if err != nil {
+		// ENOENT as error means that node is not (yet) part of the ceph cluster
+		if strings.Contains(string(buf), "ENOENT") {
```

Comment:
What about this instead?

```
if code, ok := exec.ExitStatus(err); ok && code == int(syscall.ENOENT) {
```

We do that in various places already.

Reply:
Good point. Thank you for the pointer.

Changed it accordingly!